src/main/java/io/logz/logback/LogzioSender.java

```diff
@@ -112,12 +116,16 @@ public LogzioSender(String logzioToken, String logzioType, int drainTimeout, int
 
         this.tasksExecutor = tasksExecutor;
 
+        throwableProxyConverter = new ThrowableProxyConverter();
+        throwableProxyConverter.setOptionList(throwableProxyConversionOptions);
+        throwableProxyConverter.start();
 
         debug("Created new LogzioSender class");
     }
 
     public void start() {
         tasksExecutor.scheduleWithFixedDelay(this::drainQueueAndSend, 0, drainTimeout, TimeUnit.SECONDS);
+        tasksExecutor.scheduleWithFixedDelay(logsBuffer::gc, 0, 30, TimeUnit.SECONDS);
```

Comment:
Remember the caveat of using scheduleWithFixedDelay - if the method you pass here exits with an exception, the scheduler will stop the scheduling. You need to wrap it with try-catch (Throwable) and log the problem of course as warning.